x-pack/plugins/osquery/public/shared_components/osquery_action/use_osquery_menu_item.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiContextMenuItem } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { useKibana } from '../../common/lib/kibana';
+
+export const ACTION_OSQUERY = i18n.translate('xpack.osquery.alerts.osqueryAlertTitle', {
+  defaultMessage: 'Run Osquery',
+});
+
+interface IProps {
+  agentId: string;
+  onClick: () => void;
+}
+
+export const useOsqueryMenuItem = ({ agentId, onClick }: IProps) => {
```

Comment:
Can you add tests for this hook?

Reply:
I simplified the logic slightly, now this function just returns an item. I don't think there would be any value in adding a test for this. I hope you don't mind :)

Reply:
👍🏾